mmv1/products/securesourcemanager/Repository.yaml

```diff
@@ -0,0 +1,170 @@
+    name: 'secure_source_manager_repository_basic'
+    primary_resource_id: 'default'
+    primary_resource_name: "fmt.Sprintf(\"tf-test-my-repository%s\",
+      context[\"random_suffix\"\
+      ])"
+    vars:
+      repository_id: 'my-repository'
+      instance_id: 'my-instance'
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'secure_source_manager_repository_initial_config'
+    primary_resource_id: 'default'
+    primary_resource_name: "fmt.Sprintf(\"tf-test-my-repository%s\",
+      context[\"random_suffix\"\
+      ])"
+    vars:
+      repository_id: 'my-repository'
+      instance_id: 'my-instance'
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'location'
+    description: |
+      The location for the Repository.
+    required: true
+    immutable: true
+    url_param_only: true
+  - !ruby/object:Api::Type::String
+    name: 'repository_id'
+    description: |
+      The name for the Repository.
+    required: true
+    immutable: true
+    url_param_only: true
+  - !ruby/object:Api::Type::String
+    name: 'instance'
+    description: |
+      The name of the instance in which the repository is hosted.
+    required: true
+    diff_suppress_func: 'tpgresource.ProjectNumberDiffSuppress'
+properties:
+  - !ruby/object:Api::Type::String
+    name: 'name'
+    description: |
+      The resource name for the Repository.
+    output: true
+    immutable: true
+  - !ruby/object:Api::Type::String
+    name: 'description'
+    description: |
+      Description of the repository, which cannot exceed 500 characters.
+  - !ruby/object:Api::Type::String
+    name: 'uid'
+    description: |
+      Unique identifier of the repository.
+    output: true
+  - !ruby/object:Api::Type::Time
+    name: 'createTime'
+    description: |
+      Time the repository was created in UTC.
+    output: true
+  - !ruby/object:Api::Type::Time
+    name: 'updateTime'
+    description: |
+      Time the repository was updated in UTC.
+    output: true
+  - !ruby/object:Api::Type::String
+    name: 'etag'
+    description: |
+      This checksum is computed by the server based on the value of other fields,
+      and may be sent on update and delete requests to ensure the client has an
+      up-to-date value before proceeding.
+  - !ruby/object:Api::Type::NestedObject
+    name: 'uris'
+    description: |
+      URIs for the repository.
+    output: true
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'html'
+        description: |
+          HTML is the URI for the user to view the repository in a browser.
+        output: true
+      - !ruby/object:Api::Type::String
+        name: 'gitHttps'
+        description:
+          git_https is the git HTTPS URI for git operations.
+        output: true
+      - !ruby/object:Api::Type::String
+        name: 'api'
+        description: |
+          API is the URI for API access.
+        output: true
+  - !ruby/object:Api::Type::NestedObject
+    name: 'initialConfig'
```

Comment:
Does this need to be immutable too or does ignore_read prevent updates to this field?

Reply:
We don't return initialConfig in our Get response, so if you try a Terraform plan afterwards, it attempts to replace the repository, because it thinks you are adding an initial config.

Reply:
If you change the field, Terraform asks if you want to replace it

Reply:
Ah, got it. SG then